solidity/test/token-distribution/TokenDistributorTest.js

```diff
@@ -0,0 +1,651 @@
+          unclaimedUnlockDuration,
+          {
+            from: thirdParty,
+          }
+        ),
+        "Ownable: caller is not the owner"
+      )
+    })
+  })
+
+  describe("claim", async () => {
+    before(async () => {
+      await tokenDistributor.allocate(
+        testData.merkle.merkleRoot,
+        testData.merkle.tokenTotal,
+        unclaimedUnlockDuration,
+        { from: owner }
+      )
+    })
+
+    after(freshDeployment)
+
+    it("transfers tokens", async function () {
+      const recipientInitialBalance = toBN(await testToken.balanceOf(recipient))
+      const destinationInitialBalance = toBN(
+        await testToken.balanceOf(claimDestination)
+      )
+
+      await tokenDistributor.claim(
+        recipient,
+        claimDestination,
+        testData.signature.v,
+        testData.signature.r,
+        testData.signature.s,
+        testData.merkle.claims[recipient].index,
+        testData.merkle.claims[recipient].amount,
+        testData.merkle.claims[recipient].proof
+      )
+
+      expect(
+        await testToken.balanceOf(recipient),
+        "invalid recipient address balance"
+      ).to.eq.BN(toBN(recipientInitialBalance))
+
+      expect(
+        await testToken.balanceOf(claimDestination),
+        "invalid destination address balance"
+      ).to.eq.BN(
+        destinationInitialBalance.add(
+          toBN(testData.merkle.claims[recipient].amount)
+        )
+      )
+    })
+
+    it("emits event", async function () {
+      expectEvent(
+        await tokenDistributor.claim(
+          recipient,
+          claimDestination,
+          testData.signature.v,
+          testData.signature.r,
+          testData.signature.s,
+          testData.merkle.claims[recipient].index,
+          testData.merkle.claims[recipient].amount,
+          testData.merkle.claims[recipient].proof
+        ),
+        "TokensClaimed",
+        {
+          index: testData.merkle.claims[recipient].index.toString(),
+          recipient: recipient,
+          destination: claimDestination,
+          amount: toBN(testData.merkle.claims[recipient].amount),
+        }
+      )
+    })
+
+    describe("when verifying destination signature", async function () {
+      async function signatureVerificationTest(
+        signerPrivateKey,
+        submitter,
+        shouldRevert
+      ) {
+        const signature = web3.eth.accounts.sign(
+          web3.utils.keccak256(claimDestination),
+          signerPrivateKey
+        )
+
+        claimFuncCall = tokenDistributor.claim(
+          recipient,
+          claimDestination,
+          signature.v,
+          signature.r,
+          signature.s,
+          testData.merkle.claims[recipient].index,
+          testData.merkle.claims[recipient].amount,
+          testData.merkle.claims[recipient].proof,
+          { from: submitter }
+        )
+
+        shouldRevert
+          ? await expectRevert(claimFuncCall, "invalid signature")
+          : await claimFuncCall
```

Comment:
I think it is better to expect something, maybe an emitted event at the end to make sure it was a successful call?